### ISSUE TITLE
Pypy installation fail on OSX

### ISSUE THREAD

**Issue by karolhor** [pyenv/pyenv#911]
Hi,
I wanted to install `pypy2-5.7.1` on my OSX 10.12.3 but I can't. Here is an output:

```bash
$ pyenv install -v pypy2-5.7.1
/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641 ~/.pyenv
Downloading pypy2-v5.7.1-osx64.tar.bz2...

05/15 12:56:04 [NOTICE] Downloading 1 item(s)

05/15 12:56:04 [ERROR] CUID#7 - Download aborted. URI=https://pyenv.github.io/pythons/4e99ba356432861534917a9477ace0ccee617bd631512759a530f8383e153a3d
Exception: [AbstractCommand.cc:351] errorCode=3 URI=https://pyenv.github.io/pythons/4e99ba356432861534917a9477ace0ccee617bd631512759a530f8383e153a3d
  -> [HttpSkipResponseCommand.cc:219] errorCode=3 Resource not found

05/15 12:56:04 [NOTICE] Download GID#73773a16e347e66d not complete:

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
73773a|ERR |       0B/s|https://pyenv.github.io/pythons/4e99ba356432861534917a9477ace0ccee617bd631512759a530f8383e153a3d

Status Legend:
(ERR):error occurred.

aria2 will resume download if the transfer is restarted.
If there are any errors, then see the log file. See '-l' option in help/man page for details.
-> https://bitbucket.org/pypy/pypy/downloads/pypy2-v5.7.1-osx64.tar.bz2

05/15 12:56:04 [NOTICE] Downloading 1 item(s)
[#ba0dba 0B/0B CN:1 DL:0B]

05/15 12:56:05 [NOTICE] Allocating disk space. Use --file-allocation=none to disable it. See --file-allocation option in man page for more details.
[#ba0dba 112KiB/18MiB(0%) CN:1 DL:212KiB ETA:1m27s]
[#ba0dba 1.5MiB/18MiB(8%) CN:1 DL:0.9MiB ETA:16s]
[#ba0dba 4.2MiB/18MiB(23%) CN:1 DL:1.6MiB ETA:8s]
[#ba0dba 6.9MiB/18MiB(38%) CN:1 DL:1.9MiB ETA:5s]
[#ba0dba 10MiB/18MiB(54%) CN:1 DL:2.1MiB ETA:3s]
[#ba0dba 12MiB/18MiB(67%) CN:1 DL:2.1MiB ETA:2s]
[#ba0dba 15MiB/18MiB(85%) CN:1 DL:2.3MiB ETA:1s]
[#ba0dba 18MiB/18MiB(99%) CN:1 DL:2.3MiB]

05/15 12:56:13 [NOTICE] Download complete: /private/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641/pypy2-v5.7.1-osx64.tar.bz2

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
ba0dba|OK  |   2.3MiB/s|/private/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641/pypy2-v5.7.1-osx64.tar.bz2

Status Legend:
(OK):download completed.
/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641/pypy2-v5.7.1-osx64 /var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641 ~/.pyenv
Installing pypy2-v5.7.1-osx64...
ERROR: invalid Python executable: /Users/karolhorowski/.pyenv/versions/pypy2-5.7.1/bin/python

The python-build could not find proper executable of Python after successful build.
Please open an issue for future improvements.
https://github.com/pyenv/pyenv/issues

BUILD FAILED (OS X 10.12.3 using python-build 20160602)

Inspect or clean up the working tree at /var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641
Results logged to /var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641.log

Last 10 log lines:
05/15 12:56:13 [NOTICE] Download complete: /private/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641/pypy2-v5.7.1-osx64.tar.bz2

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
ba0dba|OK  |   2.3MiB/s|/private/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641/pypy2-v5.7.1-osx64.tar.bz2

Status Legend:
(OK):download completed.
/var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641/pypy2-v5.7.1-osx64 /var/folders/8v/6rnfcxrd16z29cbc5v6g40yctt1vg0/T/python-build.20170515125604.96641 ~/.pyenv
```

It looks like `python-builder` can't find `python` in `bin` directory. In fact, there shouldn't be there .

I tested this also with other pypy versions:
- pypy2-5.7.0
- pypy2-5.6.0
- pypy2-5.3
- pypy-5.0.0

and always I've get same error.

I have the newest `pyenv` from master branche: `32922007863c4a54feca2a95226c8307cfdfea3d`

**Comment by andrew-christianson** [comment]
@karolhor quite a bit later, but: did you happen to have `--enable-framework` in `PYTHON_CONFIGURE_OPTIONS` when this happened?

**Comment by karolhor** [comment]
@andrew-christianson I haven't set any `PYTHON_CONFIGURE_OPTIONS` variable but only used default options. After creating this issue I haven't check if this error still exists. Now I have `pyenv` from newest `master` branch (commit a2d51328eeef85b108303a31d90a8b0ba923a8ef) and it works! 

I've tested installing versions:
- `pypy2-5.7.1`,
- `pypy2.7-6.0.0`,
- `pypy3.5-6.0.0`

I've also tested installation with `PYTHON_CONFIGURE_OPTS="--enable-framework"` and it fails like in #1155.

I think this issue can be close because defaults works and fix issues in #1155
